compiler/hash-typecheck/src/error.rs

```diff
@@ -1,25 +1,374 @@
+                ty_def_name,
+                field_name,
+                location: field_location,
+            } => {
+                let name = IDENTIFIER_MAP.get_ident(field_name);
+                let ty_name = IDENTIFIER_MAP.get_ident(ty_def_name);
+
+                // If we have the location of the definition, we can print it here
+                if let Some(ty_def_location) = ty_def_location {
+                    builder.add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        ty_def_location,
+                        format!("The struct `{}` is defined here.", ty_name),
+                    )));
+                }
+
+                builder
+                    .add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        field_location,
+                        "Struct is missing field",
+                    )))
+                    .add_element(ReportElement::Note(ReportNote::new(
+                        "note",
+                        format!("The struct `{}` is missing the field `{}`.", ty_name, name),
+                    )));
+            }
+            TypecheckError::BoundRequiresStrictlyTypeVars(location) => {
+                // @@TODO: Maybe report here what we found?
+                builder
+                    .add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        location, "here",
+                    )))
+                    .add_element(ReportElement::Note(ReportNote::new(
+                        "note",
+                        "This type bound should only contain type variables",
+                    )));
+            }
+            TypecheckError::InvalidPropertyAccess {
+                field_name,
+                location,
+                ty_def_name,
+                ty_def_location,
+            } => {
+                let name = IDENTIFIER_MAP.get_ident(field_name);
+                let ty_name = IDENTIFIER_MAP.get_ident(ty_def_name);
+
+                // If we have the location of the definition, we can print it here
+                if let Some(ty_def_location) = ty_def_location {
+                    builder.add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        ty_def_location,
+                        format!("The struct `{}` is defined here.", ty_name),
+                    )));
+                }
+
+                builder
+                    .add_element(ReportElement::CodeBlock(ReportCodeBlock::new(
+                        location,
+                        "unknown field",
+                    )))
+                    .add_element(ReportElement::Note(ReportNote::new(
+                        "note",
+                        format!("The field `{}` doesn't exist on type `{}`.", name, ty_name),
+                    )));
+            }
+        }
+
+        // @@ErrorReporting: we might want to properly handle incomplete reports?
+        builder.build().unwrap()
```

Comment:
When would such a thing happen?

Reply:
This was just a copied over a comment from the `ParseError` implementation.